### DISCUSSION TITLE
Service Install Rollback Issue

### DISCUSSION THREAD

**Discussion by knguyen99** [wixtoolset/issues#8779] [Questions]
Hi, 

 im having an issue when performing a failedupgrade and rollback occurs, the service that was previously installed does not restart and cannot start from services application. The service is an apache procrun service. 

i can install the service manually again by clicking on the service gui exe located in the install directory and installing but cannot do it automatically through the msi.

After much debugging i found that the issue is that this registry key is missing when trying to reinstall the service: HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\[service name]\Parameters

The registry is an empty key defined in the installer with ForceCreateOnInstall = yes. During rollback, I do not see this registry key being created. Is there anyway to make sure it gets created on rollback too?

**Comment by barnson** [comment]
"SetupBuilder" isn't a WiX thing.

**Reply by knguyen99** [reply]
it’s a gradle plugin that builds wix files. https://github.com/i-net-software/SetupBuilder

**Reply by chrpai** [reply]
There are a number of third party tools that emit wxs source and build it to create an MSI.  They are usually narrowly focused for a specific usecase and frequently create installers that don't follow best practices.  The WiX team cannot support these and as a setup consultant I don't use them because I prefer general purpose authoring tools such as my very own IsWiX.  

Take a look at my tutorial.  You'll find creating an MSI for a service is easy. 

https://github.com/iswix-llc/iswix-tutorials

Paid development and support services are available from various sources.

**Reply by knguyen99** [reply]
i really just need help on figuring out why an empty registry key isn’t being created on rollback

**Reply by wmanning** [reply]
I can't help you but you'll probably need to provide a minimum reproducible sample using the WiX toolset alone before anyone would really take a look. Or at least provide the WiX authoring.
